my_game.py

```diff
@@ -131,7 +145,8 @@ def on_show_view(self):
         self.pe.add_collision_handler(
             "player",
             "object",
-            post_handler = self.handler_player_object,
+            # Has to be begin_handler or pre_handler to make avoid collision possible
+            begin_handler = self.handler_player_object,
```

Comment:
'pre_handler' is _Two shapes are touching during this step_
'begin_handler' is _Two shapes just started touching for the first time this step_

I think you are doing the correct thing here.
https://www.pymunk.org/en/latest/_modules/pymunk/collision_handler.html